controllers/clustergroupupgrade_controller.go

```diff
@@ -435,9 +437,9 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 				}
 				// Set completion time only after post actions are executed with no errors
 				clusterGroupUpgrade.Status.Status.CompletedAt = metav1.Now()
+				nextReconcile = doNotRequeue()
```

Comment:
It should be doNotRequeue already. Are you trying to get rid of excessive reconcile calls post completion?